spacesuit/src/bit_range.rs

```diff
@@ -0,0 +1,37 @@
+/// Represents a usize with value in the range [0,64]
+#[derive(Copy, Clone, Debug)]
+pub struct BitRange(usize);
+
+impl BitRange {
+    /// Returns Some(BitRange) if `n` is ≤ 64.
+    /// Otherwise returns None.
+    pub fn new(n: usize) -> Option<Self> {
+        if n > 64 {
+            None
+        } else {
+            Some(BitRange(n))
+        }
+    }
+
+    /// Returns 64-bit range
+    pub fn max() -> Self {
+        BitRange(64)
+    }
+
+    /// Converts BitRange to a `u8`
+    pub fn to_u8(self) -> u8 {
+        self.0 as u8
+    }
+
+    /// Checks whether the Bitrange lies
+    /// between 0 and 64
+    pub fn is_valid(&self) -> bool {
```

Comment:
Opinion: this method is unnecessary on philosophical grounds. Validity of the BitRange must be encoded _in the type_ (see also `Tx` vs `ValidTx`). We already make sure it's impossible to construct BitRange with invalid integer, so any BitRange instance is automatically a valid one.

That's the whole point of having a separate type instead of `usize`. So we can avoid repetitive checks all over the place that can be sometimes forgotten. "Encode protocol requirements in the type system" and all that :-)

Reply:
got it, I thought maybe someone could do BitRange(65) somewhere up the line

Reply:
oh right:

```
            Opcode::Range => {
                let bit_width =
                    BitRange::new(program.read_u8()? as usize).ok_or(VMError::FormatError)?;
                Ok(Instruction::Range(bit_width))
            }
```